test/e2e-shared-tests/mcd_config_drift.go

```diff
@@ -218,6 +231,62 @@ func (c configDriftTest) Run(t *testing.T) {
 				})
 			},
 		},
+		{
+			name:           "update degraded reason on drift",
+			rebootExpected: false,
+			testFunc: func(t *testing.T) {
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err := c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+
+				firstReason := node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey]
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, configDriftFileContents)
+				assertNodeAndMCPIsRecovered(t, c.ClientSet, c.node, c.mcp)
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilenameTwo, "incorrect data 2")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilenameTwo)
```

Comment:
To speed things up, you can remove this `assertNodeAndMCPIsDegraded` in favor of using the one below. In other words, mutate the two files on the node, and then wait for the degradation status to be reported.